acvm-repo/acvm/benches/arithmetic_solver.rs

```diff
@@ -48,6 +48,25 @@ fn perfectly_parallel_opcodes(c: &mut Criterion) {
     bench_bytecode(c, "perfectly_parallel_opcodes", &bytecode);
 }
 
+fn perfectly_parallel_batch_inversion_opcodes(c: &mut Criterion) {
+    // This bytecode defines a set of constraints such that `w_{i+1} = 2*w_0 + i`.
+    // This allows all opcodes to be solved with a single field inversion assuming perfect batching.
+    let bytecode: Vec<Opcode<FieldElement>> = (1..1000)
+        .map(|witness_index| {
+            Opcode::AssertZero(Expression {
+                mul_terms: Vec::new(),
+                linear_combinations: vec![
+                    (FieldElement::one(), Witness(0)),
+                    (-FieldElement::from(2_u32), Witness(witness_index)),
+                ],
+                q_c: FieldElement::from(witness_index),
```

Comment:
Does this need to be incremented for `w_{i+1}`?
```suggestion
                q_c: FieldElement::from(witness_index+1),
```